### DISCUSSION TITLE
Is it possible to have a dynamic collection name? So that I can use the same Schema for multiple collections?

### DISCUSSION THREAD

**Discussion by dotKokott** [BeanieODM/beanie#849] [Question] [answered]
I have multiple collections that use the same schema.

Is it somehow possible to initialise a document multiple times with different settings? 
Or what else would be the proper way to define a schema that refers to multiple collections?

**Comment by dlbrittain** [comment]
I would like to bump this as I am curious as well.

**Comment by dotKokott** [comment, marked as answer]
Found a solution by now. Feels like a workaround but it works
```
class DatasetItem(Document):
  pass

async def get_dataset_item_model(
    db_name: str,
    collection_name: str,
) -> type[DatasetItem]:
    class DynamicDatasetItem(DatasetItem):
        class Settings:
            name = collection_name

    init_beanie(db=db_name, document_models=[DynamicDatasetItem])

    return DynamicDatasetItem

model1 = await get_dataset_item_model("DB1", "Collection1")
model2 = await get_dataset_item_model("DB2", "Collection2")

await model2.get(...)
```

**Reply by frenzyk** [reply]
This is very bad solution.
1) Every time you call `get_dataset_item_model` new dynamic class will be created and this is costly. Also forget about pypy JIT.
2) In big project it will very hard to find usages of exact collection names. Code will be poorly self-explanatory. 
It can be made better by dynamically creating and using classes once, during app start. But it will only eliminate class creation costs and necessity to write function for each action.
